### PR TITLE
Created checkstyle for java files

### PR DESCRIPTION
**Pull request by haram-dev**
#602 I've added maven checkstyle for java files.
Although I've confirmed that checking styles of the files go well according to the requirements, I might have made some mistakes because I'm almost new to open source.
Any comments and suggestions are welcome.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -751,6 +754,29 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-project-info-reports-plugin</artifactId>
+                <version>${maven.project.info.reports.plugin.version}</version>
+            </plugin>
         </plugins>
     </build>
+
+    <reporting>
+        <plugins>
```
Comment (bibinss) [inline pom.xml]: we need to move this under build->plugins
also add a goal to the plugin
```
        <plugin>
                <groupId>org.apache.maven.plugins</groupId>
                <artifactId>maven-checkstyle-plugin</artifactId>
                <version>${maven.checkstyle.plugin.version}</version>
                <configuration>
                    <configLocation>checkstyle.xml</configLocation>
                </configuration>
                <executions>
                    <execution>
                        <goals>
                            <goal>check</goal>
                        </goals>
                    </execution>
                </executions>
            </plugin>```

File: pom.xml [hunk 2]
```diff
@@ -751,6 +754,29 @@
...
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-site-plugin</artifactId>
+                <version>${maven.site.plugin.version}</version>
+            </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-project-info-reports-plugin</artifactId>
+                <version>${maven.project.info.reports.plugin.version}</version>
+            </plugin>
```
Comment (bibinss) [inline pom.xml]: not sure we really need these plugins. why are these added?

File: pom.xml (hunk 1 above)
Comment (bibinss) [reply pom.xml]: once you add this, you will see some checkstyle breakages and you can fix them also as part of this PR

**Comment by bibinss** [comment]
@haram-dev will you be able to work on my suggestions here^^ ?

**Comment by bibinss** [comment]
since I don't see a response/update here and since another dev has raised a new PR for this, I am going ahead with the new PR.

**Comment by williewheeler** [comment]
PR #612 by @0001vrn supersedes this PR.
